src/Cookies.php

```diff
@@ -197,7 +199,7 @@ public function getMatchingCookies(
      * @param string $cookieName The cookie's name
      * @param int $retAs Whether to return cookies as objects of \Zend\Http\Header\SetCookie or as strings
      * @throws Exception\InvalidArgumentException if invalid URI specified or invalid $retAs value
-     * @return SetCookie|string
+     * @return SetCookie|string|false
```

Comment:
`false` is not a type: `bool` is to be used here

Reply:
[phpdoc](http://docs.phpdoc.org/guides/types.html#types-of-types) says:

> A type can either be
>
> - a class name, either Fully Qualified or aliases; such as \DateTime or Entity (with a namespace alias elsewhere in the same source file).
> - a keyword for a primitive in PHP, such as int or string.
> - a special keyword specific to the PHPDoc Standard, such as false or mixed.

So it should be a valid value

Reply:
I was looking around. Since PHP 7 `false` is internally a type.

There is a discussion in a phpstan [issue](https://github.com/phpstan/phpstan/issues/43#issuecomment-269314362).

Another resource of Nikic is this [article](https://nikic.github.io/2015/05/05/Internal-value-representation-in-PHP-7-part-1.html).

Reply:
`false` is not a type: it is a value of the type `bool`, and hinting on a value is not something that tooling reading types should nor does deal with.

Whether PHP internally optimises `false` is a different (unrelated) story

Reply:
Ok, I'm going to change it using `bool`, but I will need to add more type checks.

Reply:
There was already `false` return types in the code. I'll try to change them.